concordium-node/src/consensus_ffi/ffi.rs

```diff
@@ -348,6 +348,19 @@ pub struct NotificationHandlers {
     pub finalized_blocks: futures::channel::mpsc::UnboundedReceiver<Arc<[u8]>>,
 }
 
+/// A type of callback used to signal the Rust code of a pending unsupported
+/// protocol update. The callback is called with:
+/// - the context
+/// - effective time of unsupported update (milliseconds since unix epoch).
+type NotifyUnsupportedUpdatesCallback =
+    unsafe extern "C" fn(*mut NotifyUnsupportedUpdatesContext, u64);
+
+pub struct NotifyUnsupportedUpdatesContext {
+    /// If non-zero the value represents the effective timestamp of unsupported
+    /// protocol update as milliseconds since unix epoch.
+    pub unsupported_pending_protocol_version: prometheus::IntGauge,
```

Comment:
Is there a specific reason why we are not using
```
GenericGauge<AtomicU64>
```
instead?

Then we don't have to do `as i64`

Reply:
The only reason is to follow the type of the other metrics with timestamp, but it would not make any difference in the output of a scrape.
I'll change it